src/js_sync.hpp

```diff
@@ -887,15 +878,16 @@ void SessionClass<T>::wait_for_completion(Direction direction, ContextType ctx,
 
         util::EventLoopDispatcher<DownloadUploadCompletionHandler> completion_handler(
             [ctx = Protected(Context<T>::get_global_context(ctx)),
-             callback = Protected(ctx, callback)](std::error_code error) {
+             callback = Protected(ctx, callback)](Status status) {
                 HANDLESCOPE(ctx);
-                Function<T>::callback(
-                    ctx, callback,
-                    {!error ? Value::from_undefined(ctx)
-                            : Object::create_obj(ctx, {
-                                                          {"message", Value::from_string(ctx, error.message())},
-                                                          {"errorCode", Value::from_number(ctx, error.value())},
-                                                      })});
+                std::vector<typename T::Value> arguments;
+                if (!status.is_ok()) {
+                    arguments.push_back(Object::create_obj(ctx, {
+                        {"message", Value::from_string(ctx, status.reason())},
+                        {"errorCode", Value::from_string(ctx, status.code_string())},
```

Comment:
Our public API docs doesn't promise anything of the error objects, but we should probably mention this in a note in the changelog anyway.

Reply:
Actually I had to revert this because just like `SyncError` the `Status` passed to the completion handler here is constructed from `std:error_code`. Seems like Sync still relies on `std:error_code` internally.